src/panels/config/entity_registry/ha-config-entity-registry.ts

```diff
@@ -192,8 +383,106 @@ class HaConfigEntityRegistry extends LitElement {
     }
   }
 
-  private _showDisabledChanged(ev: Event) {
-    this._showDisabled = (ev.target as HaSwitch).checked;
+  private _showDisabledChanged() {
+    this._showDisabled = !this._showDisabled;
+  }
+
+  private _showRestoredChanged() {
+    this._showRestored = !this._showRestored;
+  }
+
+  private _handleSearchChange(ev: CustomEvent) {
+    this._filter = ev.detail.value;
+  }
+
+  private _handleSelectionChanged(ev: CustomEvent): void {
+    const changedSelection = ev.detail as SelectionChangedEvent;
+    const entity = changedSelection.id;
+    if (changedSelection.selected) {
+      this._selectedEntities.push(entity);
+    } else {
+      const index = this._selectedEntities.indexOf(entity);
+      if (index !== -1) {
+        this._selectedEntities.splice(index, 1);
+      }
+    }
+    this._selectedEntities = [...this._selectedEntities];
+  }
+
+  private _enableSelected() {
+    showConfirmationDialog(this, {
+      title: this.hass.localize(
+        "ui.panel.config.entity_registry.picker.enable_selected.confirm_title",
+        "number",
+        this._selectedEntities.length
+      ),
+      text: this.hass.localize(
+        "ui.panel.config.entity_registry.picker.enable_selected.confirm_text"
+      ),
+      confirmBtnText: this.hass.localize("ui.common.yes"),
+      cancelBtnText: this.hass.localize("ui.common.no"),
+      confirm: () => {
+        this._selectedEntities.forEach((entity) =>
+          updateEntityRegistryEntry(this.hass, entity, {
+            disabled_by: null,
+          })
+        );
+        this._clearSelection();
+      },
+    });
+  }
+
+  private _disableSelected() {
+    showConfirmationDialog(this, {
+      title: this.hass.localize(
+        "ui.panel.config.entity_registry.picker.disable_selected.confirm_title",
+        "number",
+        this._selectedEntities.length
+      ),
+      text: this.hass.localize(
+        "ui.panel.config.entity_registry.picker.disable_selected.confirm_text"
+      ),
+      confirmBtnText: this.hass.localize("ui.common.yes"),
+      cancelBtnText: this.hass.localize("ui.common.no"),
+      confirm: () => {
+        this._selectedEntities.forEach((entity) =>
+          updateEntityRegistryEntry(this.hass, entity, {
+            disabled_by: "user",
+          })
+        );
+        this._clearSelection();
+      },
+    });
+  }
+
+  private _removeSelected() {
+    showConfirmationDialog(this, {
+      title: this.hass.localize(
+        "ui.panel.config.entity_registry.picker.remove_selected.confirm_title",
+        "number",
+        this._selectedEntities.length
```

Comment:
Can we already filter out the entities that we're going to delete here, and maybe even name the ones that we can't delete right now?